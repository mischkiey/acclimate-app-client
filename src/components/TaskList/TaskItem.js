import React from 'react';

function TaskItem(props) {
    return (
        <>  
            <div className='list-group'>
                <button
                    className='y-btn'
                    onClick={(e) => props.handleCheckTask(e, props.user_id, props.user_task_id, props.user_task_completed)}
                >
                    <i className="material-icons">check</i>
                </button>
                <input
                    className=
                        {
                            (props.user_task_completed)
                                ? 'completed item-input' 
                                : 'item-input'
                        }
                    defaultValue={props.user_task}
                    onChange={(e) => {props.handleEditTask(e, props.user_id, props.user_task_id)}}
                    type='text'
                />
                <button
                    className='r-btn'
                    onClick={() => props.handleDeleteTask(props.user_task_id)}
                >
                    <i className="material-icons">delete</i>
                </button>
            </div>    
        </>
    );

};

export default TaskItem;