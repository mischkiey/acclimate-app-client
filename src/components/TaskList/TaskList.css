@import url('https://fonts.googleapis.com/icon?family=Material+Icons');

.list-group {
    display: flex;
    flex-flow: row;
    justify-content: center;
}