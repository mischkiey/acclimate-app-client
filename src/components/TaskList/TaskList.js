import React, { Component } from 'react';

class TaskList extends Component {
    render () {
        return (
            <>
                <h2>Task List</h2>
            </>
        );
    };
};

export default TaskList;