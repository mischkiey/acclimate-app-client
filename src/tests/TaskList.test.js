import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter, Router } from 'react-router-dom';
import TaskList from '../components/TaskList/TaskList';

describe(`TaskList Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>    
                    <TaskList />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <TaskList />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})