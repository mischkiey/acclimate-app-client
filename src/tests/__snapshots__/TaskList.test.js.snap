// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Component Snapshot test Renders the UI as expected 1`] = `
Array [
  <h2
    className="center"
  >
    Task List
  </h2>,
  <form
    className="list-group"
    onSubmit={[Function]}
  >
    <input
      className="item-input"
      name="add_user_task"
      placeholder="Enter task"
      type="text"
    />
    <button
      className="y-btn"
    >
      <i
        className="material-icons"
      >
        add_circle
      </i>
    </button>
  </form>,
]
`;
